### ISSUE TITLE
OLS-autocomplete doesn't bring any of its own dependencies

### ISSUE THREAD

**Issue by tschaka1904** [LLTommy/OLS-autocomplete#4]
Hi, 

I was just playing the with `OLS-autocomplete`, but received a few errors whilst setting it up. It seems that the `OLS-autocomplete` doesn't bring any of its dependencies? I've got errors about `Bloodhound`, `handlebars` or `Typeahead` are not defined.

**Comment by LLTommy** [comment]
Maybe I missed to add the dependencies, I will look into it later today!

**Comment by tschaka1904** [comment]
Thanks! Please let me know once it is ready to try.

**Comment by LLTommy** [comment]
I just had a look. Yes I did not define some dependencies. If you look at the examples htmls, I include them via external links - you could do the same for now to test it!  

For jQuery for example, not to include it makes sense, because many pages include jQuery already so I don't want to package it into the autocomplete widget. But you are right, of course you could make a very strong case of including the dependencies that you mentioned - since they most likely are not included by other plugins.

**Comment by tschaka1904** [comment]
I agree with what you say about jQuery. We don't use any of the other dependencies (Complex-Portal), but it would be good if the `OLS-autocomplete` would bring them. The advantage for us, for example, would be that we use `webpack` which is doing some tree shaking on our dependencies. This would not be possible if we would add them to the global scope using a `<script>`-tag.

**Comment by LLTommy** [comment]
Well, as I said, you have a very good point. What I mean by this: Yes I will change it and include the dependencies, it makes a lot of sense! I will change that asap.

**Comment by LLTommy** [comment]
Hi there, I had a look into this. The thing is that bloodhound and typeahead via npm behaves a bit funky (see https://github.com/twitter/typeahead.js/issues/1546) - let's call it a bug. There is a workaround, some people say, but I couldn't make it work yet.  

This could be the reason why I did not include the dependencies in first place. I will try to find some more time this week to look into that workaround.

**Comment by tschaka1904** [comment]
Can you give an estimate when/if could be solved?

**Comment by tschaka1904** [comment]
Can you give an estimate when/if could be solved?

**Comment by tschaka1904** [comment]
Can you give an estimate when/if could be solved?
As we would like to integrate it into [www.ebi.ac.uk/complexportal/](http://www.ebi.ac.uk/complexportal/).

**Comment by LLTommy** [comment]
Hi there. Well, if you study the link, the problem is they did not address this problem, so their is no obvious solution. The thing some people suggest is to manually edit some lines in the import. However, as far as I understand, people would that include that manually added file .... which would not help you with your web pack question, right? 

One guy suggested :

>  I worked around it by importing typeahead.jquery.js and bloodhound-js separately

I did not find the *right* packages that he talks about, but I'll try again.

**Comment by LLTommy** [comment]
There is also a discussion on stackoverflow about this, here https://stackoverflow.com/questions/30750916/how-to-reference-typeahead-and-bloodhound-when-loading-npm-typeahead-js but I couldn't make it work so far. Potentially the answers refer to older version on npm or I am missing something. :/
